temporalcli/commands.workflow.go

```diff
@@ -262,6 +266,96 @@ func (c *TemporalWorkflowSignalCommand) run(cctx *CommandContext, args []string)
 	return nil
 }
 
+func (c *TemporalWorkflowSignalWithStartCommand) run(cctx *CommandContext, _ []string) error {
+	if c.SharedWorkflowStartOptions.WorkflowId == "" {
+		return fmt.Errorf("--workflow-id flag must be provided")
+	}
+
+	cl, err := c.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	wfStartOpts, err := buildStartOptions(&c.SharedWorkflowStartOptions, &c.WorkflowStartOptions)
+	if err != nil {
+		return err
+	}
+	wfInput, err := c.buildRawInputPayloads()
+	if err != nil {
+		return err
+	}
+
+	signalPayloadInputOpts := PayloadInputOptions{
+		Input:       c.SignalInput,
+		InputFile:   c.SignalInputFile,
+		InputMeta:   c.InputMeta,
+		InputBase64: c.SignalInputBase64,
+	}
+	signalInput, err := signalPayloadInputOpts.buildRawInputPayloads()
+	if err != nil {
+		return err
+	}
+
+	var retryPolicy *common.RetryPolicy
+	if wfStartOpts.RetryPolicy != nil {
+		retryPolicy = &commonpb.RetryPolicy{
+			MaximumInterval:        durationpb.New(wfStartOpts.RetryPolicy.MaximumInterval),
+			InitialInterval:        durationpb.New(wfStartOpts.RetryPolicy.InitialInterval),
+			BackoffCoefficient:     wfStartOpts.RetryPolicy.BackoffCoefficient,
+			MaximumAttempts:        wfStartOpts.RetryPolicy.MaximumAttempts,
+			NonRetryableErrorTypes: wfStartOpts.RetryPolicy.NonRetryableErrorTypes,
+		}
+	}
+	var memo *common.Memo
+	if wfStartOpts.Memo != nil {
+		fields, err := encodeMapToPayloads(wfStartOpts.Memo)
+		if err != nil {
+			return err
+		}
+		memo = &common.Memo{Fields: fields}
+	}
```

Comment:
I think this kind of logic needs to match `temporal workflow start` (which uses `stringKeysJSONValues` for memo). In fact, looking at this in general, is there a reason not to use the Go SDK's high-level `SignalWithStartWorkflow` call instead of the raw RPC? Then you can just reuse the `buildStartOptions` helper that exists. Also I would consider moving this into `commands.workflow_exec.go` with its other start workflow friends.

Reply:
There's a comment above the workflow service call below - `SignalWithStartWorkflow` only accepts a single arg for signal. The existing signal command on CLI uses the workflow service call because of this. I think it makes sense to provide the flexibility for multiple signal args and stay consistent with the existing CLI signal command

Reply:
I see, that is unfortunate that Go SDK is that way but I agree this needs to support multiple signal arguments. I fear that people that implement a new start option may not know to come update this. But I cannot think of a better way short of having Go SDK support a signal-with-start that takes multiple signal parameters. @Quinn-With-Two-Ns - thoughts here?

I do think it still may make more sense to move that `encodeMapToPayloads` utility that used to be for schedules only into this file for more generic use, and maybe move this command impl to `commands.workflow_exec.go` next to the other commands that work with the workflow start options.

Reply:
`signal-with-start` moved to `commands.workflow_exec.go` and `encodeMapToPayloads` to `commands.go`